File: ECommerceFront/src/app/Models/products-dto.ts
import { ResourceModel } from "../helper/resource-model";
import { BrandsDto } from "./brands-dto";
import { Category } from "./category";

export class ProductsDto extends ResourceModel<ProductsDto> {
  ProductNameAr!: string;
  ProductNameEn !: string;
  Description !: string;
  Price!: number;
  Quantity!: number;
  BrandId !: number;
  CategoryId!: number;
  brand!: BrandsDto;
  category!: Category;
}

File: ECommerceAPI/Models/ProductsDto.cs
namespace ECommerceAPI.Models
{
    public class ProductsDto
    {
        public long? Id { get; set; }
        public string ProductNameAr { get; set; }
        public string ProductNameEn { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public long BrandId { get; set; }
        public long CategoryId { get; set; }

    }
}
